tests/Infrastructure.Tests/Contracts/RecentContractRepositoryTests.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Linq;
+using Application.Contracts;
+using Domain.Contracts;
+using Domain.Users;
+using Infrastructure.Databases;
+using Microsoft.Extensions.Logging;
+
+namespace Infrastructure.Tests.Contracts;
+
+[Collection("DatabaseTests")]
+public class RecentContractRepositoryTests : IClassFixture<TestDatabaseFixture>, IDisposable
+{
+    private readonly EFRecentContractRepository _cut;
+    private readonly EFDatabaseContext _context;
+
+    public RecentContractRepositoryTests(TestDatabaseFixture fixture)
+    {
+        _context = fixture.CreateContext();
+        _cut = new EFRecentContractRepository(_context, Mock.Of<ILogger<EFRecentContractRepository>>());
+        _context.Database.BeginTransaction();
+    }
+
+    [Fact]
+    public void AddRecent_AddsTheContractCorrectly()
+    {
+        // Arrange
+        var user1 = new User() { Name = "User1" };
+        AddUserToContext(user1);
+
+        var contract1 = new Contract() { Name = "1" };
+        _context.Contracts.Add(contract1);
+
+        // Act
+        _cut.AddRecent(user1.Name, contract1);
+        IList<RecentlyViewedContract> contracts = _cut.FetchRecentContracts(user1.Name);
+
+        // Assert
+        contracts.First().ContractId.Should().Be(contract1.Id);
+    }
+
+    [Fact]
+    public void AddRecent_ReAddingAnExistingContractUpdatesTheTimeCorrectly()
```

Comment:
nit:
It's not a hard rule that you have to append the name of the method under test, you don't have to force it in like this. 

```suggestion
    public void AddingContract_UpdatesTheTime_WhenContractAlreadyExists()
```

That way you can split it up into a very clear three-part structure which tells the reader `WhatIsBeingTested_WhatIsExpectedToHappen_UnderWhichConditions`.